c/meterpreter/source/metsrv/pool_party.c

```diff
@@ -0,0 +1,299 @@
+#include "common.h"
+#include "pool_party.h"
+#include "pool_party_ext.h"
+
+pNtDll *ntdll = NULL;
+POOLPARTY_INJECTOR* poolLifeguard = NULL;
+
+pNtDll* GetOrInitNtDll() {
+	BOOL bError = FALSE;
+	HANDLE hHeap = GetProcessHeap();
+	bError = (hHeap == NULL);
+
+	if (ntdll != NULL) {
+		return ntdll;
+	}
+
+	if (!bError) {
+		ntdll = (pNtDll*)HeapAlloc(hHeap, HEAP_ZERO_MEMORY, sizeof(pNtDll));
+		bError = ntdll == NULL;
+		if (!bError) {
+			HMODULE hNtDll = LoadLibraryA("ntdll.dll");
+			ntdll->pNtQueryInformationProcess = (NTSTATUS(NTAPI*)(HANDLE, PROCESSINFOCLASS, PVOID, ULONG, PULONG))GetProcAddress(hNtDll, "NtQueryInformationProcess");
+			ntdll->pNtQueryObject = (NTSTATUS(NTAPI*)(HANDLE, OBJECT_INFORMATION_CLASS, PVOID, ULONG, PULONG))GetProcAddress(hNtDll, "NtQueryObject");
+			dprintf("[INJECT][inject_via_poolparty][ntdll_init] NtQueryInformationProcess: %p NtQueryObject: %p", ntdll->pNtQueryInformationProcess, ntdll->pNtQueryObject);
+			
+			ntdll->pZwSetIoCompletion = (NTSTATUS(NTAPI*)(HANDLE, PVOID, PVOID, NTSTATUS, ULONG_PTR))GetProcAddress(hNtDll, "ZwSetIoCompletion");
+			if (ntdll->pZwSetIoCompletion != NULL) {
```

Comment:
Is there a reason we're checking that `pZwSetIoCompletion` is set correctly, but not `pNtQueryInformationProcess` or `pNtQueryObject`?

Reply:
Because in **theory** we should always have `NtQueryInformationProcess` and `NtQueryObject`, I can add additonal checks.